django/test/utils.py

```diff
@@ -113,21 +113,25 @@ def setup_test_environment(debug=None):
     if debug is None:
         debug = settings.DEBUG
 
-    _SavedSettings.debug = settings.DEBUG
-    settings.DEBUG = debug
+    # Storing previous values in the settings module itself is problematic.
+    # See #20636.  We store them in a private class instead.  Also, we use
+    # a Namespace object for convenience because it creates new attributes
+    # on assignment.
+    saved_data = Namespace()
```

Comment:
What about using `types.SimpleNamespace` on Python 3 and defining `class SimpleNamespace(object): pass` for Python 2? Or doing only the latter for both PY2/3 with a todo? Seeing an `argparse` import in this code is a tad confusing, I think.